### PR TITLE
Improve results plotter

### PR DESCRIPTION
**Pull request by Pastafarianist**
Fixes two issues with `results_plotter` I ran into: crash because of too large episode window and mandatory use of `num_timesteps`. Please see commit messages and/or diffs for details.

### CODE REVIEW

**Comment by araffin** [comment]
Hello,
thank you for your PR.
Please do not forget to read the contributing guide and to update the changelog ;)

**Comment by Pastafarianist** [comment]
Done. Please review.

File: docs/misc/changelog.rst [hunk 1]
```diff
@@ -20,6 +20,8 @@ Pre-Release 2.5.1a0 (WIP)
   ``set_attr`` now returns ``None`` rather than a list of ``None``. (@kantneel)
 - ``GAIL``: ``gail.dataset.ExpertDataset` supports loading from memory rather than file, and
   ``gail.dataset.record_expert`` supports returning in-memory rather than saving to file.
+- fixed bug where result plotter would crash on very short runs
```
Comment (araffin) [inline docs/misc/changelog.rst]: Please add your username at the end of the entry (see above), and add it at the bottom to the list of contributors ;)

File: stable_baselines/results_plotter.py [hunk 2]
```diff
@@ -84,8 +84,9 @@ def plot_curves(xy_list, xaxis, title):
     for (i, (x, y)) in enumerate(xy_list):
         color = COLORS[i]
         plt.scatter(x, y, s=2)
-        x, y_mean = window_func(x, y, EPISODES_WINDOW, np.mean)  # So returns average of last EPISODE_WINDOW episodes
-        plt.plot(x, y_mean, color=color)
+        if x.shape[0] >= EPISODES_WINDOW:
```
Comment (araffin) [inline stable_baselines/results_plotter.py]: Missing comment

**Comment by Pastafarianist** [comment]
Addressed comments and amended commit history.

**Comment by araffin** [comment]
Ok, please fix codacy complain (line too long, and we can merge ;))

**Comment by Pastafarianist** [comment]
Uh, I'm not sure that this Codacy complaint is relevant. Is following up on its warnings a strict policy?

**Comment by araffin** [comment]
> Is following up on its warnings a strict policy?

as written in the contribution guide: "A PR must pass the Continuous Integration tests (travis + codacy) to be merged with the master branch."
So, yes, this ensure to keep the code quality. Here, if you put the comment above the line, this should pass.

**Pull request merged**